gammapy/spectrum/dataset.py

```diff
@@ -398,6 +409,66 @@ def create(cls, e_reco, e_true=None, reference_time="2000-01-01"):
         )
 
 
+    def stack(self, other):
+        """Stack this dataset with another one.
+
+        Safe mask is applied to compute the stacked counts vector.
+        Counts outside each dataset safe mask are lost.
+
+        Stacking is performed in-place.
+
+        Parameters
+        ----------
+        other : `~gammapy.spectrum.SpectrumDataset`
+            the dataset to stack to the current one
+       """
+
+        if not isinstance(other, SpectrumDataset):
+            raise TypeError("Incompatible types for SpectrumDataset stacking")
+
+        if self.counts is not None:
+            self.counts.data[~self.mask_safe] *= 0
+            self.counts.data[other.mask_safe] += other.counts.data[other.mask_safe]
+        else:
+            self.counts = None
```

Comment:
If counts is already `None` I think there is no need to set it to `None` again?

Reply:
Indeed